### PR TITLE
[GLM4.1V and GLM4.5V] Add vision transformer num_dummy_head support: max tp=4 -> max tp=8 

### PR DESCRIPTION
**Pull request by byjiang1996**
## Motivation

In some GLM4v models, `num_key_value_heads=8` in text attention and `num_heads=12` in vision attention so `tp=4` is the max supported tp_size today `(num_key_value_heads % tp_size == 0 && num_heads % tp_size == 0)`

In sglang vision attention, it supports `num_dummy_head` to help provide dummy attention head for padding. After this change, `tp=8` works for GLM now with `num_dummy_heads=4`: `(num_key_value_heads % tp_size == 0 && (num_dummy_heads + num_heads) % tp_size == 0)`
## Modifications

Followed the `num_dummy_head` padding logic in internvl: https://github.com/sgl-project/sglang/blob/f003cd354868a4c442adb710a81678fcaa47b0eb/python/sglang/srt/models/internvl.py#L562

In addition, store both mmmu model answer and final eval output val to two different files for debugging

## Accuracy Tests

Pass GLM test_vision_openai_server_b UT under tp=8, tp=4 and tp=1

mmmu score stays the same as before

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/models/glm4v.py [hunk 1]
```diff
@@ -537,6 +539,51 @@ def get_video_feature(self, items: List[MultimodalDataItem]) -> torch.Tensor:
...
+        elif "attn.proj.weight" in name:
+            padded_weight = loaded_weight.new_zeros(
+                loaded_weight.shape[0], head_dim * num_dummy_heads
+            )
+            loaded_weight = torch.cat([loaded_weight, padded_weight], dim=-1)
+        elif "attn.q_norm.weight" in name or "attn.k_norm.weight" in name:
+            padded_weight = loaded_weight.new_zeros(head_dim * num_dummy_heads)
+            loaded_weight = torch.cat([loaded_weight, padded_weight], dim=0)
+        return loaded_weight
+
```
Comment (JustinTong0323) [inline python/sglang/srt/models/glm4v.py:586]: These utility functions seem universal and could also be applied to other ViTs, could you move them to a appropriate  utils.py file?

Comment (byjiang1996) [reply python/sglang/srt/models/glm4v.py:586]: Good point. Moved to vision_utils.py

**Review by JustinTong0323 — APPROVED** [review]
Good job!
